spring-cloud-commons/src/main/java/org/springframework/cloud/commons/util/MismatchVerifierAutoConfiguration.java

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package org.springframework.cloud.commons.util;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.SpringBootVersion;
+import org.springframework.boot.autoconfigure.AutoConfigureOrder;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+
+/**
+ * {@link EnableAutoConfiguration Auto-configuration} that fails the build fast for incompatible
+ * versions of dependencies (e.g. invalid version of Boot).
+ *
+ * @author Marcin Grzejszczak
+ * @since 1.3.6
+ */
+@Configuration
+@ConditionalOnProperty(value = "spring.cloud.util.mismatch-verifier.enabled", matchIfMissing = true)
```

Comment:
Not sure I like the name. Maybe `spring.cloud.boot.version-verifier`? I you don't like the boot part since it's a composite, maybe just `spring.cloud.version-verifier`?

Reply:
I didn't want to actually mention the `version` in the name cause right now we're checking the version but maybe, in the future, we will be able to add more stuff there (I don't know what exactly)

Reply:
Mismatch-verifier is awkward. Mismatch is a negative and verifier is a positive. I don't actually know what that means.

config verifier would work but config is overloaded.

Just looking at @ryanjbaxter's comment on docs.

`spring.cloud.compatibility-verifier`

Reply:
Updated